app/views/nodes/index.html.slim

```diff
@@ -1,12 +1,65 @@
-h1
-  | Bootstrap cluster
+.alert.alert-info.alert-dismissible role="alert"
+  button.close[type="button" data-dismiss="alert" aria-label="Close"]
+    span[aria-hidden="true"]
+      | &times;
+  i.fa.fa-4x.pull-left aria-hidden="true"
+  span
+    | After choosing the master and clicking "Apply" all other nodes will be set to the minion role
+h1 Nodes
+.row
+  .col-xs-12.discovery-control
+    span #{Minion.count} Nodes found
+    a.btn.btn-default[href="/nodes"] Refresh
+= form_tag(nodes_path, method: "put")
+.nodes-container
+  table.table
+    thead
+      tr
+        th
+          | Id
+        th
+          | Status
+        th
+          | Hostname
+        th
+          | Role
+        th
+          | Highstate
+        th
+          | Master
+    tbody
+      - Minion.all.each do |m|
+        tr
+          th
+            | #{m.id}
+          td
+            - if m.role
+              i.fa.fa-check-circle-o.text-success.fa-2x aria-hidden="true"
+            - else
+              span.fa-stack aria-hidden="true"
+                i.fa.fa-circle.fa-stack-2x.text-success aria-hidden="true"
+                i.fa.fa-refresh.fa-stack-1x.fa-spin.fa-inverse aria-hidden="true"
+          td
+            | #{m.hostname}
+          td
+            | #{m.role}
+          td
+            | #{m.highstate}
+          td
+            - if m.role == "master"
+              = radio_button_tag "roles[#{m.hostname}]", ["master"], true, disabled: true
```

Comment:
makes sense.

> 'm not sure but I think we will also need to add a hidden field which will be an array of the workers (minion ids) currently visible. We will need to use this in order to determine the workers that will get the minion role.

this is handled automatically, as the default is always minion